cpp/src/strings/datetime.cu

```diff
@@ -68,7 +68,7 @@ enum timestamp_parse_component {
     TP_ARRAYSIZE   = 8
 };
 
-struct format_item
+struct alignas(16) format_item
```

Comment:
16-byte alignment seems wasteful. This struct has size sizeof(bool) + 3. sizeof(bool) is implementation defined, but on most compilers I think the total size of the struct is 4 bytes. So an array of these structs will be 75% overhead with `alignas(16)`.

If you replace the bool with an explicitly sized enum class of 1 byte, and then use `alignas(4)`.